pkg/cmd/cli/cmd/set/env.go

```diff
@@ -115,6 +119,133 @@ func keyToEnvName(key string) string {
 	return strings.ToUpper(validEnvNameRegexp.ReplaceAllString(key, "_"))
 }
 
+func getRefObject(f *clientcmd.Factory, reference string) (runtime.Object, error) {
+	mapper, typer := f.Object(false)
+	cmdNamespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return nil, err
+	}
+	b := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		ResourceTypeOrNameArgs(true, reference).
+		Flatten()
+	one := false
+	infos, err := b.Do().IntoSingular(&one).Infos()
+	if err != nil {
+		return nil, err
+	}
+	if len(infos) != 1 {
+		return nil, fmt.Errorf("expected a single result")
+	}
+	return infos[0].Object, nil
+}
+
+var secretStore = make(map[string]*kapi.Secret)
+
+func getSecretRefValue(f *clientcmd.Factory, secretSelector *kapi.SecretKeySelector) (string, error) {
+	secret, ok := secretStore[secretSelector.Name]
+	if !ok {
+		obj, err := getRefObject(f, fmt.Sprintf("secret/%s", secretSelector.Name))
+		if err != nil {
+			return "", err
+		}
+		secret, ok = obj.(*kapi.Secret)
+		if !ok {
+			return "", fmt.Errorf("expected type secret")
+		}
+		secretStore[secretSelector.Name] = secret
+	}
+	for key, data := range secret.Data {
```

Comment:
wow, yes.  not sure what i was thinking on this one (and the next).  maps... what a thing!